### PR TITLE
[SYCLomatic] Enable migration of more thrust APIs

### PR DESCRIPTION
**Pull request by tomflinda**
The thrust APIs below are supported to be migrated:
thrust::minmax_element
thrust::is_sorted
thrust::partition
thrust::remove_copy
thrust::unique_copy
thrust::transform_exclusive_scan
thrust::set_difference
thrust::set_difference_by_key
thrust::set_intersection_by_key
thrust::stable_sort
thrust::tabulate
thrust::for_each_n

### CODE REVIEW

**Review by AndyCHHuang — APPROVED** [review]
LGTM

File: clang/test/dpct/thrust-algo.cu [hunk 1]
```diff
@@ -13,10 +13,14 @@
 #include <thrust/extrema.h>
 #include <thrust/gather.h>
```
Comment (zhimingwang36) [inline clang/test/dpct/thrust-algo.cu:14]: can the migrated code can be build by icx?

File: clang/test/dpct/thrust-algo-raw-ptr-noneusm.cu [hunk 2]
```diff
@@ -0,0 +1,438 @@
+// UNSUPPORTED: cuda-8.0
```
Comment (zhimingwang36) [inline clang/test/dpct/thrust-algo-raw-ptr-noneusm.cu:1]: can the migrated code can be build by icx?

**Review by zhimingwang36 — CHANGES_REQUESTED** [review]
Need more raw data to see migrated code can be build or not.

File: clang/test/dpct/thrust-algo.cu (hunk 1 above)
Comment (tomflinda) [reply clang/test/dpct/thrust-algo.cu:14]: Yes，https://github.com/oneapi-src/SYCLomatic-test/pull/178 has been submitted to verify the build&run status.

File: clang/test/dpct/thrust-algo-raw-ptr-noneusm.cu (hunk 2 above)
Comment (tomflinda) [reply clang/test/dpct/thrust-algo-raw-ptr-noneusm.cu:1]: Yes，https://github.com/oneapi-src/SYCLomatic-test/pull/178 has been submitted to verify the build&run status.

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**